src/Entity/DatasetSubmission.php

```diff
@@ -1020,7 +1020,7 @@ public function __construct(Entity $entity, PersonDatasetSubmissionDatasetContac
             $this->setDatasetFileColdStorageArchiveSha256Hash($entity->getDatasetFileColdStorageArchiveSha256Hash());
             $this->setDatasetFileColdStorageArchiveSize($entity->getDatasetFileColdStorageArchiveSize());
             $this->setDatasetFileColdStorageOriginalFilename($entity->getDatasetFileColdStorageOriginalFilename());
-
+            $this->setFileset($entity->getFileset());
```

Comment:
Since this is collection, we have to copy it. Like the foreach things below. The file entities inside may be copied.
I'll look into this, to see if we can make it work.